cupy/core/core.pyx

```diff
@@ -1847,8 +1847,12 @@ cpdef ndarray dot(ndarray a, ndarray b, ndarray out=None):
         b = rollaxis(b, b_axis, 0)
 
     k = a._shape[0]
-    m = b.size // k
-    n = a.size // k
+    if k != 0:
```

Comment:
Could you write a comment that the objective of this change is to align the behavior to NumPy' one.